### PR TITLE
Developing

### PR DESCRIPTION
**Pull request by lts-hmms**
2.5

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,67 +1,200 @@
...
+        user.username = req.params.username;
+        res.status(201).send('Username changed to ' + user.username)
+} else {
+    res.status(404).send('User not found.')
+}})
+
+// deregister user
+app.delete('/users/:username',(req,res)=>{
+    res.send('Your user was deleted!')
+})
```
Comment (oxenprogrammer) [inline index.js]: May we please implement this delete endpoint?

File: index.js [hunk 2]
```diff
@@ -1,67 +1,200 @@
...
+    if (user.favorites.indexOf(newMovie) === -1){
+        user.favorites.push(newMovie)
+        res.status(201).send('Movie added to your Favslist')
+    } else {
+        res.send('Movie is already in your Favslist')
+    }})
+
+// remove movie from favorites
+app.delete('/users/:username/favorites/:title',(req,res)=>{
+    res.send('Movie removed from Favslist.')
```
Comment (oxenprogrammer) [inline index.js]: Same here

File: index.js [hunk 3]
```diff
@@ -1,67 +1,200 @@
...
+app.get('/movies/:title',(req,res)=>{
+    res.json(movies.find((movie)=>{
+        return movie.title ===req.params.title
+    }))
+})
+
+// get genre description by name
+app.get('/genres/:name',(req,res)=>{
+    res.send('This is a description about the film genre ' + req.params.name)
+})
```
Comment (oxenprogrammer) [inline index.js]: Can we get the description of a genre correctly please rather than hard coding?

File: index.js [hunk 4]
```diff
@@ -1,67 +1,200 @@
...
+
+// get director data by name
+app.get('/directors/:name',(req,res)=>{
+    res.send('This is the data about ' + req.params.name)
+})
```
Comment (oxenprogrammer) [inline index.js]: Same here

File: index.js [hunk 5]
```diff
@@ -1,67 +1,200 @@
...
+        id: "2",
+        email:  'peep@gmx.de',
+        favorites:[]
+    }
+];
+let movies = [
     {
         title: 'Black Panther',
-        director: ['Ryan Coogler']
+        director: 'Ryan Coogler',
```
Comment (oxenprogrammer) [inline index.js]: IMO a movie can have one or more directors and a director can direct in one or many movies. So the relationship can be one to many or many to many. To simplify this, let's model a relationship of `one to many` here?

File: index.js [hunk 6]
```diff
@@ -1,67 +1,200 @@
...
+        year: 2018,
+        genre: ['Action', 'Adventure', 'Fantasy'],
```
Comment (oxenprogrammer) [inline index.js]: genre as well should be modelled like 
```
genre: [
    {name: "action", description: "fighting etc" },
    {name: "romance", description: "love etc" },
    {name: "thriller", description: "crime etc" }
    ]
```

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Please help study[ this resource ](https://redis.com/blog/nosql-data-modeling/#:~:text=What%20is%20a%20NoSQL%20data,how%20it%20all%20connects%20together.)on what data modelling is and make the changes below accordingly. Thank you

File: index.js (hunk 2 above)
Comment (lts-hmms) [reply index.js]: done

File: index.js (hunk 1 above)
Comment (lts-hmms) [reply index.js]: done

File: index.js (hunk 5 above)
Comment (lts-hmms) [reply index.js]: done

File: index.js (hunk 6 above)
Comment (lts-hmms) [reply index.js]: done

File: index.js (hunk 3 above)
Comment (lts-hmms) [reply index.js]: done

File: index.js (hunk 4 above)
Comment (lts-hmms) [reply index.js]: done

File: index.js [hunk 7]
```diff
@@ -1,67 +1,256 @@
...
+
+// get specific movie
+app.get('/movies/:title',(req,res)=>{
+    res.json(movies.find((movie)=>{
+        return movie.title ===req.params.title
+    }))
+})
+
+// get genre description by name
+app.get('/movies/:title/genre/:name/description',(req,res)=>{
```
Comment (oxenprogrammer) [inline index.js:161]:
```suggestion
app.get('/movies/:title/genre/:name',(req,res)=>{
```

**Review by oxenprogrammer — APPROVED** [review]

**Pull request merged**